project-koku/koku
# AWS report filter/query_by tags

**Overview**
This feature works the same was as OCP where reports can filtered and group by with `tag:<tag_key>` 

For example:
```
http://localhost:8000/api/v1/reports/inventory/aws/instance-type/?filter[tag:environment]=dev
http://localhost:8000/api/v1/reports/inventory/aws/instance-type/?group_by[tag:environment]=*
```

**Testing:**
1.  Add static nise report data to include EC2 instances with labels:
2.  Add AWS-local provider to ingest the above nise data
3.  Ingest with masu
4.  Verify that AWS tags are returned from tag API:
5.  Issue /instance-type endpoint with no query parameters.  Verify that the response is correct:
6.  Filter the /instance-type with a filter on tag:environment for all values, correct value, and incorrect value
7.  Filter the /instance-type with a filter on tag:version for all values, correct value, and incorrect value
8.  group_by tag:version on the /instance-type endpoint
9.  group_by tag:environment on the /instance-type endpoint

**Testing Results**
[koku_aws_tag_api_ut.txt](https://github.com/project-koku/koku/files/2809150/koku_aws_tag_api_ut.txt)
